### PR TITLE
Update AGP and JAVA Version to make it compatible with Gradle 8.7 and Add support for Streaming

### PR DESCRIPTION
**Pull request by luckycreationsindia**
The version of Android Gradle Plugin (AGP) has been updated from 7.4.2 to 8.5.1 in the gradle.properties file. Additionally, the source and target Java compatibility have been changed from version 17 to version 21 in the build.gradle file.

This will fix issue with incompatible Java Version which was 17

### CODE REVIEW

**Comment by luckycreationsindia** [comment]
Sorry for multiple commits. Forgot to change version in example file and also in workflow (Using Java 21)

File: example/android/app/build.gradle [hunk 1]
```diff
@@ -30,8 +30,8 @@ android {
     compileSdkVersion 34
 
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_17
-        targetCompatibility JavaVersion.VERSION_17
+        sourceCompatibility JavaVersion.VERSION_1_8
```
[the remark this answers is not in the document]
Comment (luckycreationsindia) [reply example/android/app/build.gradle:33]: I'll give an update soon. Problem is I'm getting issue like Java Version 21 and Kotlin Version 17. I tried fixing it but the error didn't compile. Maybe I need to clean everything and try again.

[the remark this answers is not in the document]
Comment (luckycreationsindia) [reply example/android/app/build.gradle]: I changed the version back to 17 and it's working

**Comment by Harry-Chen** [comment]
Sorry for the late reply.

I'm reluctant to upgrade AGP version, since AGP 7.4.2 is the latest version that does not enforce so-called [`agpVersionChecker`](https://android.googlesource.com/platform/tools/base/+/84d514f5eba4f488f60e2d5670788f0405ddc18b). This might cause some trouble to our users, especially those using other flutter plugin that requires older AGP. 

And then, there are complex dependencies between gradle, AGP, Java and Kotlin version. I did not find a good solution to this yet.

**Comment by Harry-Chen** [comment]
As for the streaming support -- it is good, but not the type of streaming we want (i.e. with foreground dispatch mode) as mentioned in #17 or #62.

Also it involves much duplicated code (which can be avoided).

**Comment by hemantbeast** [comment]
Just need to set AGP to version 8.5.2 & below compileOptions flag need to add this flag & it'll fix the build issue.

```
kotlinOptions {
    jvmTarget = '17'
}
```

**Comment by larssn** [comment]
Thing is, the JVM target should actually be Java 8, in order to support the lower SDK levels (Android 7.1, but you can go lower with desugaring).

I'm pretty sure my comment above is wrong about this, and this should be possible:

```groovy
compileSdk 35
    
compileOptions {
    sourceCompatibility JavaVersion.VERSION_1_8
    targetCompatibility JavaVersion.VERSION_1_8
}

kotlinOptions {
    jvmTarget = '1.8'
}
```

I've just lowered these today for most of our internal plugins, and they build just fine with AGP 8.7.1.

**Comment by Harry-Chen** [comment]
I have now upgraded AGP to 8.7.0 in the latest master commit. You may want to try it (and remove the changes from your PR).

As for the streaming support, @dangfan is also actively working on it. Maybe you can have more discussion before we merge that.

**Pull request closed without merging**